### PR TITLE
chore: adaptations for nightly-2024-04-30

### PR DESCRIPTION
**Pull request by kim-em**
This will not actually build until later today, when `nightly-2024-04-30` comes out: I've been working on this with a local toolchain.

Nevertheless I'd like to start looking at the diff now.

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/AffineScheme.lean [hunk 1]
```diff
@@ -245,6 +245,8 @@ theorem isBasis_basicOpen (X : Scheme) [IsAffine X] :
 
 namespace IsAffineOpen
 
+-- Adaptation note: 2024-04-23
+-- TODO check if this `.{u}` restricts anything
 variable {X Y : Scheme.{u}} {U : Opens X} (hU : IsAffineOpen U) (f : X.presheaf.obj (op U))
```
Comment (eric-wieser) [inline Mathlib/AlgebraicGeometry/AffineScheme.lean:250]: I guess this can now be removed, assuming this was checked in review?

Comment (semorrison) [reply Mathlib/AlgebraicGeometry/AffineScheme.lean:250]: Done.

File: test/observe.lean [hunk 2]
```diff
@@ -2,12 +2,7 @@ import Mathlib.Data.Nat.Prime
 
 open Nat
 
--- Adaptation note: at nightly-2024-03-27,
--- we had to increase `maxHeartbeats` here from 8000 to 16000.
--- Adaptation note: at nightly-2024-04-01,
--- we had to increase `maxHeartbeats` here from 16000 to 24000.
--- (Note: successive runs are faster, because of the `exact?` cache initialisation.)
-set_option maxHeartbeats 24000 in
+set_option maxHeartbeats 8000 in
```
Comment (jcommelin) [inline test/observe.lean:5]: This makes me happy! :octopus:

File: Mathlib/NumberTheory/RamificationInertia.lean [hunk 3]
```diff
@@ -447,9 +446,13 @@ noncomputable instance Quotient.algebraQuotientPowRamificationIdx : Algebra (R 
   Quotient.algebraQuotientOfLEComap (Ideal.map_le_iff_le_comap.mp le_pow_ramificationIdx)
 #align ideal.quotient.algebra_quotient_pow_ramification_idx Ideal.Quotient.algebraQuotientPowRamificationIdx
 
+-- Adaptation note: 2024-04-23
+-- The right hand side here used to be `Ideal.Quotient.mk _ (f x)`, but this is now slow without
+-- `set_option backward.isDefEq.lazyProjDelta false in`
+-- Instead we've replaced it with `Ideal.Quotient.mk (P ^ e) (f x)` (compare #12412)
```
Comment (MichaelStollBayreuth) [inline Mathlib/NumberTheory/RamificationInertia.lean:452]: Just a comment: This was already slow previously, only not *that* slow. (Similarly in the analogous case below, where it was not quite as slow as here.)

File: Mathlib/Probability/Independence/Basic.lean [hunk 4]
```diff
@@ -396,6 +397,14 @@ section FromMeasurableSpacesToSetsOfSets
...
 /-! ### Independence of measurable space structures implies independence of generating π-systems -/
 
+set_option backward.synthInstance.canonInstances false in-- Adaptation note: 2024-04-28
+-- The change to typeclass resolution in
+-- https://github.com/leanprover/lean4/pull/4003
+-- (See also https://github.com/leanprover/lean4/issues/3996)
+-- will hopefully significantly speed up typeclass search in Mathlib.
+-- However it causes some breakages.
+-- Currently, we're using the backwards compatibility flag to disable the new behaviour
+-- as locally as possible, and leaving the task of cleaning this up for later.
```
Comment (MichaelStollBayreuth) [inline Mathlib/Probability/Independence/Basic.lean:407]: Why is there a long explanation here and not elsewhere?

File: Mathlib/AlgebraicGeometry/AffineScheme.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply Mathlib/AlgebraicGeometry/AffineScheme.lean:250]: Do we need to keep this as an adaptation note? These were already quite slow before.

Comment (semorrison) [reply Mathlib/AlgebraicGeometry/AffineScheme.lean:250]: Given that it was already changed on master, let's say no? I'm not exactly sure what we would add.

File: Mathlib/Probability/Independence/ZeroOne.lean [hunk 5]
```diff
@@ -21,6 +21,16 @@ is measurable with respect to the tail σ-algebra `limsup s atTop` has probabili
...
+-- Adaptation note: 2024-04-28
+-- The change to typeclass resolution in
+-- https://github.com/leanprover/lean4/pull/4003
+-- (See also https://github.com/leanprover/lean4/issues/3996)
+-- will hopefully significantly speed up typeclass search in Mathlib.
+-- However it causes some breakages.
+-- Currently, we're using the backwards compatibility flag to disable the new behaviour
+-- as locally as possible, and leaving the task of cleaning this up for later.
+-- However for this file we need it for nearly every declaration.
+set_option backward.synthInstance.canonInstances false
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Probability/Independence/ZeroOne.lean:33]: Nevertheless I'll tidy up this comment to be clearer.

**Review by kmill — APPROVED** [review]
bors d+

**Pull request merged**